ffi/mupdf.lua

```diff
@@ -655,14 +642,15 @@ function mupdf.renderImage(data, size, width, height)
     end
     local bbtype
     local ncomp = M.fz_pixmap_components(context(), pixmap)
-    if ncomp == 2 then bbtype = BlitBuffer.TYPE_BB8A
+    if ncomp == 1 then bbtype = BlitBuffer.TYPE_BB8
+    elseif ncomp == 2 then bbtype = BlitBuffer.TYPE_BB8A
+    elseif ncomp == 3 then bbtype = BlitBuffer.TYPE_BBRGB24
     elseif ncomp == 4 then bbtype = BlitBuffer.TYPE_BBRGB32
     else error("unsupported number of color components")
     end
     local p = M.fz_pixmap_samples(context(), pixmap)
     local bb = BlitBuffer.new(p_width, p_height, bbtype, p):copy()
     M.fz_drop_pixmap(context(), pixmap)
-    M.fz_drop_image(context(), image)
```

Comment:
This was already done earlier or is it no longer necessary?

Reply:
It's done earlier, as suggested by poire-z, so the error handling is cleaner.